src/test/operators/table_scan_between_test.cpp

```diff
@@ -57,50 +57,91 @@ class TableScanBetweenTest : public TypedOperatorBaseTest {
     _data_table_wrapper = std::make_shared<TableWrapper>(data_table);
     _data_table_wrapper->execute();
   }
+
+  void _test_between_boundaries(std::vector<std::tuple<AllTypeVariant, AllTypeVariant, std::vector<int>>>& tests,
+                                bool left_inclusive = true, bool right_inclusive = true) {
+    const auto& [data_type, encoding, nullable] = GetParam();
+    std::ignore = encoding;
+    resolve_data_type(data_type, [&, nullable = nullable](const auto type) {
+      for (const auto& [left, right, expected_with_null] : tests) {
+        SCOPED_TRACE(std::string("BETWEEN ") + std::to_string(boost::get<double>(left)) +
+                     (left_inclusive ? " (inclusive)" : " (exclusive)") + " AND " +
+                     std::to_string(boost::get<double>(right)) + (right_inclusive ? " (inclusive)" : " (exclusive)"));
+
+        auto scan =
+            create_between_table_scan(_data_table_wrapper, ColumnID{0}, left, right, left_inclusive, right_inclusive);
+        scan->execute();
+
+        const auto& result_table = *scan->get_output();
+        auto result_ints = std::vector<int>{};
+        for (const auto& chunk : result_table.chunks()) {
+          const auto segment_b = chunk->get_segment(ColumnID{1});
+          for (auto offset = ChunkOffset{0}; offset < segment_b->size(); ++offset) {
+            result_ints.emplace_back(boost::get<int>((*segment_b)[offset]));
+          }
+        }
+        std::sort(result_ints.begin(), result_ints.end());
+
+        auto expected = expected_with_null;
+        if (nullable) {
+          // Remove the positions that should not be included because they are NULL
+          expected.erase(std::remove_if(expected.begin(), expected.end(), [](int x) { return x % 3 == 2; }),
```

Comment:
``x % 3 == 2`` how does this work?
Maybe this could be changed to something more readable.

Reply:
okay, I have added a comment to it. Once again, it was copied/moved from its original place
Anyway, this code just removes approximately every third value. It just generates test data